### PR TITLE
[WIP] Bulk Order

### PR DESCRIPTION
**Pull request by petro97**
Description:

Linting:

- [x] No linting errors

Tests:

- [ ] E2E tests (npm test run with `e2e`)
- [x] Manual tests

Documentation:

- [x] Requires documentation updates

### CODE REVIEW

**Comment by shaunmaharaj** [comment]
@petro97 please fix the merge conflict

**Comment by shaunmaharaj** [comment]
@petro97 please ensure you ran the e2e tests

File: src/components/bulkorder.main.jsx [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+                {intl.get('bulk-order-title')}
+              </a>
+            </li>
+          </ul>
+          <div className="tab-content">
+            <div className="tab-pane fade show active" id="quick-order" role="tabpanel" aria-labelledby="quick-order-tab">
+              <div className="form-content form-content-submit col-sm-offset-4">
+                <button
+                  className="ep-btn primary small btn-itemdetail-addtocart"
+                  id="product_display_item_add_to_cart_button"
```
Comment (shaunmaharaj) [inline src/components/bulkorder.main.jsx]: duplicate id's used
```suggestion
                  id="quick_order_item_add_to_cart_button"
```

File: src/components/bulkorder.main.jsx [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+                {items.map((item, i) => (
+                  <QuickOrderForm item={item} index={i} onItemSubmit={updatedItem => this.quickFormSubmit(updatedItem, i)} />
+                ))}
+              </div>
+            </div>
+            <div className="tab-pane fade" id="bulk-order" role="tabpanel" aria-labelledby="bulk-order-tab">
+              <div className="form-content form-content-submit col-sm-offset-4">
+                <button
+                  className="ep-btn primary small btn-itemdetail-addtocart"
+                  id="product_display_item_add_to_cart_button"
```
Comment (shaunmaharaj) [inline src/components/bulkorder.main.jsx]: duplicate id's used
```suggestion
                  id="bulk_order_item_add_to_cart_button"
```

File: src/components/appheader.main.jsx [hunk 3]
```diff
@@ -185,7 +208,7 @@ class AppHeaderMain extends React.Component {
               <span className="icon glyphicon glyphicon-align-justify" />
             </button>
           </div>
-
+          <BulkOrderMain isBulkModalOpened={isBulkModalOpened} handleClose={this.handleBulkModalClose} cartData={cartData} />
```
Comment (shaunmaharaj) [inline src/components/appheader.main.jsx]: We should only show this if b2b is enabled. Even though the API is available on b2c as well, we want to only demonstrate it in b2b.

File: src/components/bulkorder.main.less [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+    .bulk-order-hide{
+      padding-left: 25px;
+      padding-top: 3px;
+      font-size: 16px;
+      color: #333333;
+    }
+  }
+
+  .bulk-modal {
+    position: relative;
```
Comment (shaunmaharaj) [inline src/components/bulkorder.main.less]: Feedback from Bonnie is to have the panel cover the entire page:

image

**Comment by BonnieEP** [comment]
@petro97
1) Mobile: missing the "hide" text beside the close "x", please add. 
2) Mobile/Desktop: overlay panel for "Quick Order/ Bulk Order " needs to be length of screen (Shaun may have already added this comment) 
3) Mobile/Desktop: Error States are missing for item SKU in Quick order tab
https://invis.io/BTQYTNXR6WV#/355072560_B2B_Bulk_Order_open_2-5 .

Error Usercases:

Invalid SKU: When the user submits a quick/bulk order with invalid SKU
Show in UI: Inline Error - "[SKUCode] is invalid. Please clear SKU to submit order"

Configurable Product: When the user submits a quick/bulk order with a configurable product
Show in UI: Inline Error - "[SKUCode] is a configurable product. Please visit the product page to add it to the cart"

Out of Stock: When the user submits a quick/bulk order with a product that is out of stock (OR has too little stock)
Show in UI: Inline Error - "Sorry this [SKUCode] only has [x] available but [y] were requested."

Duplicate SKU: When the user submits a quick/bulk order with a product SKU that is a duplicate
Show in UI: Inline Error - "There are multiple entries of [SKUCode] in the form. Please combine duplicate SKUs to the one entry."

Dependent Product: When the user submits a quick/bulk order with a product SKU that is a dependent product (bundled)
Show in UI: Inline Error - "[SKUCode] cannot be purchased independently, please visit the product page to add it to the cart."

Products Already in Cart: When the user submits a quick/bulk order with a product with the same products that are in cart

Show in UI: Pop up confirmation dialog - "Shopping Cart already contains '1' item(s). Do you still want to add item(s) to cart?" [Yes button] [No button]

If "Yes"add additional items to cart. If "No"return user to quick/bulk order form

**Comment by sminnath** [comment]
Can we have an error message for wrong SKUs in bulk order and quick order?

Also, it is not easy to find the options for Quick order and bulk order in the UI

**Comment by sminnath** [comment]
You have used SKU and item number in : 
Copy and paste a product SKU and quantity separated by a comma. Make sure each new item number and quantity is on a separate line and in the following format:
be consistent, either use SKU or item number 
**Enter product SKU and quantity separated by a comma. Ensure that each entry is in separate line in the following format:**



Copy and Paste in input below

Change this to:

File: src/components/bulkorder.main.less [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+
+          &:hover {
+            border-color: transparent;
+            color: @mainTextColor;
+          }
+
+          &.active {
+            color: @mainTextColor;
+            font-weight: bold;
+            border-bottom: 3px solid #0099DE;
```
Comment (shaunmaharaj) [inline src/components/bulkorder.main.less]: Can we generalize this color in common.less?

**Comment by shaunmaharaj** [comment]
@petro97 As part of this, lets also remove the quick order form on the Cart Page (since it's being replaced with this feature). May require deleting `QuickOrderMain` as well as a component (../components/quickorder.main)

**Comment by shaunmaharaj** [comment]
No further comments. Thanks @petro97!!

File: src/components/quickorderform.jsx [hunk 6]
```diff
@@ -0,0 +1,260 @@
...
+            }
+          })
+          .catch((error) => {
+            onItemSubmit({
+              code, quantity, product: {}, isValidField: false,
+            });
+            this.setState({
+              skuErrorMessage: `${productId} ${intl.get('sku-invalid-message')}`,
+            });
+            console.error(error.message);
```
Comment (shaunmaharaj) [inline src/components/quickorderform.jsx]: This will get flagged as a linting error.

File: src/localization/fr-FR.json [hunk 7]
```diff
@@ -21,6 +21,9 @@
   "i-have-an-account": "I- -h-a-v-e- -a-n- -a-c-c-o-u-n-t",
   "registration-error": "E-r-r-o-r- -r-e-g-i-s-t-e-r-i-n-g- -y-o-u-r- -a-c-c-o-u-n-t-.",
   "invalid-username-or-password": "Y-o-u-r- -u-s-e-r-n-a-m-e- -o-r- -p-a-s-s-w-o-r-d- -i-s- -i-n-v-a-l-i-d-.",
+  "sku-invalid-message": "i-s- -i-n-v-a-l-i-d-.- -P-l-e-a-s-e- -c-l-e-a-r- -S-K-U- -t-o- -s-u-b-m-i-t- -o-r-d-e-r",
```
Comment (shaunmaharaj) [inline src/localization/fr-FR.json]: FYI @sminnath @BonnieEP 
Any concerns with these messages?

**Pull request merged**